pkg/rdkafka/RdKafkaContext.php

```diff
@@ -96,7 +93,23 @@ public function createTemporaryQueue(): Queue
      */
     public function createProducer(): Producer
     {
-        return new RdKafkaProducer($this->getProducer(), $this->getSerializer());
+        if (!isset($this->producer)) {
+            $producer = new VendorProducer($this->getConf());
+
+            if (isset($this->config['log_level'])) {
+                $producer->setLogLevel($this->config['log_level']);
```

Comment:
if this `setLogLevel` of `RdKafka\Producer` be aware that it has been deprecated

Reply:
@nick-zh you're correct, this is directly calling `setLogLevel` on Kafka Producer instance.

At this point I'm not really going to remove this deprecation, since in general PHP frameworks and applications should only log the deprecation (similarly to the default topic case). I'll address this in a separate PR.